java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -6080,6 +6075,68 @@ void testAllFilteredFromValidity() {
     }
   }
 
+  @Test
+  void testRemoveNullMasksIfNeeded() {
+    ListType nestedType = new ListType(true, new StructType(false,
+        new BasicType(true, DType.INT32),
+        new BasicType(true, DType.INT64)));
+
+    List data1 = Arrays.asList(10, 20L);
+    List data2 = Arrays.asList(50, 60L);
+    HostColumnVector.StructData structData1 = new HostColumnVector.StructData(data1);
+    HostColumnVector.StructData structData2 = new HostColumnVector.StructData(data2);
+
+    //First we create ColumnVectors
+    try (ColumnVector nonNullVector0 = ColumnVector.fromBoxedInts(1, 2, 3);
+         ColumnVector nonNullVector2 = ColumnVector.fromStrings("1", "2", "3");
+         ColumnVector nonNullVector1 = ColumnVector.fromLists(nestedType,
+             Arrays.asList(structData1, structData2),
+             Arrays.asList(structData1, structData2),
+             Arrays.asList(structData1, structData2))) {
+      //Then we take the created ColumnVectors and add validity masks even though the nullCount = 0
+      try (ColumnVector cv0 = new ColumnVector(nonNullVector0.type, nonNullVector0.rows,
+          Optional.of(0L),
+          new DeviceMemoryBuffer(nonNullVector0.getData().address,
+              nonNullVector0.getData().length, Cuda.DEFAULT_STREAM),
+          DeviceMemoryBuffer.allocate(BitVectorHelper.getValidityAllocationSizeInBytes(nonNullVector0.rows)),
+          nonNullVector0.getOffsets() == null ? null :
+              new DeviceMemoryBuffer(nonNullVector0.getOffsets().address,
+                  nonNullVector0.getOffsets().length,
+                  Cuda.DEFAULT_STREAM));
+           ColumnVector cv1 = new ColumnVector(nonNullVector1.type, nonNullVector1.rows,
+               Optional.of(0L),
+               null,
+               DeviceMemoryBuffer.allocate(BitVectorHelper.getValidityLengthInBytes(nonNullVector1.rows)),
```

Comment:
This is uninitialized memory. We probably want to do a mem-set on it so there is no question what is in it.